hadoop-hdfs-project/hadoop-hdfs-client/src/main/proto/ClientNamenodeProtocol.proto

```diff
@@ -428,6 +428,14 @@ message GetPreferredBlockSizeResponseProto {
 message GetSlowDatanodeReportRequestProto {
 }
 
+message GetEnclosingRootRequestProto {
+  required string filename = 1;
```

Comment:
I'd use optional. Required has been deprecated in Proto3. 

https://stackoverflow.com/questions/31801257/why-required-and-optional-is-removed-in-protocol-buffers-3

Reply:
@steveloughran Any thoughts here? most of the protos seems to use required, but @mkuchenbecker does raise a good point using `optional` for better forward compatibility. Does hadoop codebase have any opinions here?

Reply:
* the use of required probably dates from the 2.x era of the current protocols.
* i don't do protobuf stuff so have no valid opinions

Reply:
sounds good. Ill update these to `optional` since it is best practice for proto going forward and hadoop doesn't seem to be super opinionated about this.